SU2_CFD/src/solvers/CNEMOEulerSolver.cpp

```diff
@@ -1873,126 +1900,36 @@ void CNEMOEulerSolver::SetNondimensionalization(CConfig *config, unsigned short
+    geometry->vertex[val_marker][iVertex]->GetNormal(Normal);
 
-  for (iVar = 0; iVar < nVar; iVar++) {
-    Jacobian_b[iVar] = new su2double[nVar];
-    DubDu[iVar] = new su2double[nVar];
-  }
-
-  /*--- Get species molar mass ---*/
-  auto& Ms = FluidModel->GetSpeciesMolarMass();
+    Area = GeometryToolbox::Norm(nDim, Normal);
 
-  /*--- Loop over all the vertices on this boundary (val_marker) ---*/
-  for (iVertex = 0; iVertex < geometry->nVertex[val_marker]; iVertex++) {
+    for(iDim = 0; iDim < nDim; iDim++)
+      UnitNormal[iDim] = Normal[iDim]/Area;
+ 
     iPoint = geometry->vertex[val_marker][iVertex]->GetNode();
 
-    /*--- Check if the node belongs to the domain (i.e, not a halo node) ---*/
-    if (geometry->nodes->GetDomain(iPoint)) {
-
-      /*--- Normal vector for this vertex (negative for outward convention) ---*/
-      geometry->vertex[val_marker][iVertex]->GetNormal(Normal);
-
-      /*--- Calculate parameters from the geometry ---*/
-      Area = GeometryToolbox::Norm(nDim, Normal);
-
-      for (iDim = 0; iDim < nDim; iDim++){
-        NormalArea[iDim] = -Normal[iDim];
-        UnitNormal[iDim] = -Normal[iDim]/Area;
-      }
-
-      /*--- Retrieve the pressure on the vertex ---*/
-      P   = nodes->GetPressure(iPoint);
-
-      /*--- Apply the flow-tangency b.c. to the convective flux ---*/
-      for (iSpecies = 0; iSpecies < nSpecies; iSpecies++)
-        Residual[iSpecies] = 0.0;
-      for (iDim = 0; iDim < nDim; iDim++){
-        Residual[nSpecies+iDim] = P * UnitNormal[iDim] * Area;
-      }
-      Residual[nSpecies+nDim]   = 0.0;
-      Residual[nSpecies+nDim+1] = 0.0;
-
-      /*--- Add value to the residual ---*/
-      LinSysRes.AddBlock(iPoint, Residual);
-
-      /*--- If using implicit time-stepping, calculate b.c. contribution to Jacobian ---*/
-      if (implicit) {
-
-        /*--- Initialize Jacobian ---*/
-        for (iVar = 0; iVar < nVar; iVar++)
-          for (jVar = 0; jVar < nVar; jVar++)
-            Jacobian_i[iVar][jVar] = 0.0;
-
-        /*--- Calculate state i ---*/
-        rho     = nodes->GetDensity(iPoint);
-        rhoE    = nodes->GetSolution(iPoint,nSpecies+nDim);
-        rhoEve  = nodes->GetSolution(iPoint,nSpecies+nDim+1);
-        dPdU    = nodes->GetdPdU(iPoint);
-        for (iDim = 0; iDim < nDim; iDim++)
-          u[iDim] = nodes->GetVelocity(iPoint,iDim);
-
-        conc = 0.0;
-        for (iSpecies = 0; iSpecies < nSpecies; iSpecies++) {
-          cs    = nodes->GetMassFraction(iPoint,iSpecies);
-          conc += cs * rho/Ms[iSpecies];
+    Residual_Old = LinSysRes.GetBlock(iPoint);
 
-          /////// NEW //////
-          for (iDim = 0; iDim < nDim; iDim++) {
-            Jacobian_i[nSpecies+iDim][iSpecies] = dPdU[iSpecies] * UnitNormal[iDim];
-            Jacobian_i[iSpecies][nSpecies+iDim] = cs * UnitNormal[iDim];
-          }
-        }
+    for(iDim = 0; iDim < nDim; iDim++)
+      Residual[nSpecies+iDim] = Residual_Old[nSpecies+iDim]*abs(UnitNormal[iDim]);
```

Comment:
The abs(UnitNormal) looks strange, do you want to reflect it about the symmetry plane?

Reply:
The goal here is to set the normal component of the (momentum-)residual to zero. That is why for now it is also done after all other residual contributions so that no other BC has the chance to add a normal residual. It might be unnecessary if no contribution downstream changes s.th. on that front so I would say it is a precaution until we know better.

@fmpmorgado Is it possible that this only works if the normal vector is pointing exactly in one coordinate direction e.g. (1,0,0)? I would say that `R = R - (R*n)R`  with `R`esdiual and unit-`n`ormal is the right way. In code:
```c++
auto NormalComponent= GeometryToolbox::DotProduct(nDim, &Residual_Old[nSpecies], UnitNormal);
for(iDim = 0; iDim < nDim; iDim++)
    Residual[nSpecies+iDim] = NormalComponent * Residual_Old[nSpecies+iDim];

LinSysRes.SubtractBlock(iPoint, Residual);
```
If I am not mistaken the UnitNormal points inward here but that doesn't even matter imo and just subtracting the Block should be fine.

Reply:
@pcarruscag You are right, already corrected it.